### PR TITLE
New Snapshot Review Dialog

### PR DESCRIPTION
**Pull request by zfox23**
**Known Issues:**

1. Making the dataweb API call to delete the user story does not delete the image on S3 as well.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3e9ec9389a445bb84e3820ee2ecf1bbee740cce4 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3e9ec93.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3e9ec93.dmg

File: scripts/system/snapshot.js [hunk 1]
```diff
@@ -170,12 +304,16 @@ function onClicked() {
...
         }, SNAPSHOT_DELAY);
     }, FINISH_SOUND_DELAY);
 }
 
 function isDomainOpen(id) {
+    print("Checking open status of domain with ID:", id);
+    if (!id) {
+        return false;
+    }
     var request = new XMLHttpRequest();
```
Comment (howard-stearns) [inline scripts/system/snapshot.js]: As long as we have request here now, let's use that for this as well. (Only simplifies slightly, but having one less pattern is always makes things easier to read. No need to try to figure out "why'd they do this one differently....?")

**Comment by howard-stearns** [comment]
test result: If you don't have a snapshot directory set, it prompts for you to pick one before taking the picture. (That's correct behavior.) But at that time it brings back the HUD/toolbar (and presumably tablet?), and then that HUD/toolbar (tablet?) is in the .gif picture. (The timing works out so that it is not in the .jpg.)

**Comment by howard-stearns** [comment]
test result: Use the settings button to change settings, and save. The BEHAVIOR of the snapshot dialog is then correct (e.g., it respects the newly changed settings), but the "CAMERA CAPTURES" radio button is not updated to reflect the change.

**Comment by howard-stearns** [comment]
test result: I took a snap, clicked share, and then cancel. But it's on my page of recent activity (https://highfidelity.com/users/howard.stearns). It isn't on other pages, so it's probably a bug on the server, but we'll have to figure that out.

**Comment by howard-stearns** [comment]
design observation (I know this doesn't implement the final design): it seems like after we share in some way, "cancel" should change to "dismiss" or "close".

**Comment by howard-stearns** [comment]
test fail (although we'll have to see where the problem is): sharing with the announce box checked, still gives a user story action of 'snapshot' , instead of 'announcement'.

File: scripts/system/snapshot.js [hunk 2]
```diff
@@ -252,57 +405,73 @@ function processingGifStarted(pathStillSnapshot) {
...
-        action: readyData
+        action: "addImages",
+        options: snapshotOptions,
+        image_data: imageData
     }));
 }
-
+function maybeDeleteSnapshotStories() {
+    if (storyIDsToMaybeDelete.length > 0) {
+        print("User took new snapshot & didn't share old one(s); deleting old snapshot stories");
```
Comment (zfox23) [inline scripts/system/snapshot.js]: Good point; I will make a pass through all the `print` and `console.log` statements and hide unnecessary ones behind a debug flag.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 45cead91fa06ef7d5289de56d461fde0721e4080 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-45cead9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-45cead9.dmg

File: scripts/system/snapshot.js [hunk 3]
```diff
@@ -252,57 +405,73 @@ function processingGifStarted(pathStillSnapshot) {
...
         type: "snapshot",
-        action: readyData
+        action: "addImages",
+        options: snapshotOptions,
+        image_data: imageData
     }));
 }
-
+function maybeDeleteSnapshotStories() {
+    if (storyIDsToMaybeDelete.length > 0) {
```
Comment (zfox23) [inline scripts/system/snapshot.js]: Good point.

File: scripts/system/snapshot.js (hunk 1 above)
Comment (zfox23) [inline scripts/system/snapshot.js]: Ah yes, good idea.

File: scripts/system/snapshot.js [hunk 4]
```diff
@@ -225,25 +373,30 @@ function stillSnapshotTaken(pathStillSnapshot, notify) {
...
+    snapshotOptions = {
+        containsGif: true,
+        processingGif: true,
+        loadingGifPath: Script.resolvePath(Script.resourcesPath() + 'icons/loadingDark.gif'),
+        canShare: !!isDomainOpen(domainId)
+    };
+    imageData = [{ localPath: pathStillSnapshot, href: href }];
+    Settings.setValue("previousStillSnapPath", pathStillSnapshot);
+
+    tablet.emitScriptEvent(JSON.stringify({
```
Comment (zfox23) [inline scripts/system/snapshot.js]: I don't know - I was following the pattern used prior to my changes. I will mark this change as low-priority and test it if I have time.

File: scripts/system/snapshot.js [hunk 5]
```diff
@@ -252,57 +405,73 @@ function processingGifStarted(pathStillSnapshot) {
...
+        options: snapshotOptions,
+        image_data: imageData
     }));
 }
-
+function maybeDeleteSnapshotStories() {
+    if (storyIDsToMaybeDelete.length > 0) {
+        print("User took new snapshot & didn't share old one(s); deleting old snapshot stories");
+        storyIDsToMaybeDelete.forEach(function (element, idx, array) {
+            request({
```
Comment (zfox23) [inline scripts/system/snapshot.js]: OK, I will add that comment :+1:

**Comment by zfox23** [comment]
Recording: Upload a SECOND user_story when announcing instead of changing the action of the first one!

**Comment by howard-stearns** [comment]
testing issue: the snap button in the toolbar doesn't reflect the app's open state.

This didn't need to work this way before, because the snap button meant "take a picture now".  But in this design, the snap means "open the snap app".  With this meaning, it should reflect the state of the window.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b31dc4c0dec7aab780b48c00dd33ca1725314f88 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-b31dc4c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-b31dc4c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b86e3fddd2284b8ce2850198a4afd0738b449c62 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-b86e3fd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-b86e3fd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1a0f5cc2ce7d892e1fb114060e751717195cc9bf available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-1a0f5cc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-1a0f5cc.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 93529ac5d0a42a845ad2919ea6e0ddd3d7da9bc1 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-93529ac.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-93529ac.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eacaf61fbca4eb5b011e6ab7562b8819163f14c6 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-eacaf61.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-eacaf61.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0241d250aa51b9d8124297b39b44e49ff1dafa9b available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-0241d25.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-0241d25.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a33779c3f4236983dd6120c61f91ed3714399efa available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-a33779c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-a33779c.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3fe1eb70f417d466e47bf304ba5e61c43f55d0ac available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3fe1eb7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3fe1eb7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: acf9ce15f2214d4e5c38d0c70f37bb9cf0ce0d4d available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-acf9ce1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-acf9ce1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d06c1ed56d30831d659fcba71cc9ae59eaf58263 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-d06c1ed.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-d06c1ed.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6a1f3847019b8605f437f9f2e0fb1bf48825d2d5 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-6a1f384.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-6a1f384.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8c8dacd90e0cf73c8b6135765d68e41b12defd7a available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-8c8dacd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-8c8dacd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc0a7143711942fafe0bdfd12ec795e604e389e0 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-bc0a714.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-bc0a714.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5e7ba86459bbc412020a5909c0eb4de60fd4b68b available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-5e7ba86.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-5e7ba86.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0877fa59b862f49d56b132376dbfc6c93d09ba54 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-0877fa5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-0877fa5.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4152e44d2c123eca7fd55500a63645d74093af02 available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-4152e44.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-4152e44.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f146fca9de0b20b35958b1216a23f2ab0cf320b available at: https://gist.github.com/3d9fb82001a7dd79630c43a809bbd063<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3f146fc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10278-3f146fc.dmg

**Comment by howard-stearns** [comment]
Zach will continue working on a large integrated test plan.
But we need to get this merged so that we can:
  - integrate other prs. can't continue bottlenecking through here
  - we can uncover issues
  - have multiple people fix them, and fix them individually
  - get a stand non-dev, non-pr build for mac (not only for mac testing, but for two-machine cases)

**Pull request merged**